COMPILING

```diff
@@ -183,6 +183,40 @@ can be used for building with MSBuild.  Note that you still need to run
 flex/bison using "make generated_files" before opening the project.
 
 
+WORKING WITH CMAKE (EXPERIMENTAL)
+---------------------------------
+
+There is an experimental build based on CMake instead of hand-written
+makefiles. You may prefer to use this, as it should work on a wider variety of
+systems than the standard makefile build, and can integrate better with IDEs
+and static-analysis tools.
+
+0) Run `cmake --version`. If you get a command-not-found error, or the installed
+   version is lower than 3.2, go and install a new version. Most Linux distros
+   have a package for CMake, and Mac users can get it through homebrew.
+   Windows users should download it manually from cmake.org.
+
+1) Navigate to the CBMC directory in a terminal.
+
+2) Create somewhere to store your build:
+   `mkdir build`
+
+3) Generate build files with cmake:
+   `cmake -H. -Bbuild`
+   This command tells cmake to use the configuration in the current directory,
+   and to generate build files into the `build` directory.
+   This is the point to specify custom build settings, such as compilers and
+   build back-ends. You can use clang (for example) by adding the argument
+   `-DCMAKE_CXX_COMPILER=clang++` to the command line. You can also tell
+   CMake to generate IDE projects by supplying the `-G` flag.
+   Run `cmake -G` for a comprehensive list of supported back-ends.
```

Comment:
Thank you for the detailed instructions. Yet: how to set compiler options? For example, how to change optimisation, how to set additional linking flags? The bits currently in config.inc must be clearly documented here.